store/clientSettings.ts

```diff
@@ -2,48 +2,79 @@ import { MutationTree, ActionTree } from 'vuex';
 import nuxtConfig from '~/nuxt.config';
 
 /**
- * Casted typings for the CustomPrefs property of DisplayPreferencesDto
+ * Mutations that should be synced to the server
  */
-export interface CustomPreferences {
+export const syncedMutations = ['SET_DARK_MODE', 'SET_LOCALE'];
+
+/**
+ * Cast typings for the CustomPrefs property of DisplayPreferencesDto
+ */
+export interface ClientSettingsState {
   darkMode: boolean;
   locale: string;
+  lastSync: number | null;
 }
```

Comment:
I understand ``XState`` stuff as stuff that will only live and be used in that specific store (see ``PlaybackManager``). However, these typings might actually be wanted in components to have suggestions and proper typings, so I'm not sure exactly the reasoning behind the added ``State`` term.

Non-blocking though.

Reply:
Because the type represents the state of the store. This makes it more explicit.

Anything that would use this type would be linked to the state itself anyway. If not, it should use its own type, since the state might change and, since it isn't linked to it, it'll break.